### PR TITLE
Adding a tx encoder to the SDK

### PR DESCRIPTION
**Pull request by mahsamoosavi**
This is a reference implementation of how to calculate tx hash for Arbitrum Nitro specific EIP-2718 types.

### CODE REVIEW

**Comment by gzeoneth** [comment]
I think some unrelated change got included here, the `getDepositTimeout` should be from another PR?

**Comment by mahsamoosavi** [comment]
> I think some unrelated change got included here, the `getDepositTimeout` should be from another PR?

Oh right! Mixed things up. There should not be **any** changes on the `L1ToL2Message` class in this PR.

**Comment by mahsamoosavi** [comment]
> > I think some unrelated change got included here, the `getDepositTimeout` should be from another PR?
> 
> Oh right! Mixed things up. There should not be **any** changes on the `L1ToL2Message` class in this PR.

[ac3495091f488b182877244f8e1a183d787c0d70](https://github.com/OffchainLabs/arbitrum-sdk/commit/ac3495091f488b182877244f8e1a183d787c0d70) would be good to revert to?

**Comment by gzeoneth** [comment]
you can hard reset to c-nitro-stable and then force push

File: src/lib/message/L1ToL2Message.ts [hunk 1]
```diff
@@ -447,6 +447,30 @@ export class L1ToL2MessageReader extends L1ToL2Message {
     return (await this.getSuccessfulRedeem()).status
   }
 
+  private getDepositTimeout(chainId: number) {
```
Comment (yahgwai) [inline src/lib/message/L1ToL2Message.ts:450]: It looks like these changes somehow got mixed in from another file

**Pull request closed without merging**